### PR TITLE
Run test case after using patch file : nginx-1.17.4-balancer_status_code.patch

### PR DESCRIPTION
**Pull request by totemofwolf**
### Issues resolved

patch file : nginx-1.17.4-balancer_status_code.patch from https://raw.githubusercontent.com/openresty/openresty/master/patches/nginx-1.17.4-balancer_status_code.patch

Fix issue : https://github.com/alibaba/tengine/issues/1381#issuecomment-541493008

### CODE REVIEW

**Comment by membphis** [comment]
@totemofwolf You need to clean the cache and run the test again. Otherwise the Travis still use the old version.

image


https://github.com/apache/incubator-apisix/blob/master/.travis/linux_tengine_runner.sh#L39

**Comment by totemofwolf** [comment]
> @totemofwolf You need to clean the cache and run the test again. Otherwise the Travis still use the old version.
> 
> image
> 
> https://github.com/apache/incubator-apisix/blob/master/.travis/linux_tengine_runner.sh#L39

choose `Delete all repo caches`?

屏幕快照 2019-11-11 下午5 47 00

File: .travis/linux_tengine_runner.sh [hunk 1]
```diff
@@ -70,9 +70,60 @@ tengine_install() {
...
+    wget -P patches https://raw.githubusercontent.com/openresty/openresty/master/patches/nginx-1.17.4-ssl_cert_cb_yield.patch
+    wget -P patches https://raw.githubusercontent.com/openresty/openresty/master/patches/nginx-1.17.4-ssl_sess_cb_yield.patch
+    wget -P patches https://raw.githubusercontent.com/openresty/openresty/master/patches/nginx-1.17.4-stream_balancer_export.patch
+    wget -P patches https://raw.githubusercontent.com/openresty/openresty/master/patches/nginx-1.17.4-stream_proxy_get_next_upstream_tries.patch
+    wget -P patches https://raw.githubusercontent.com/openresty/openresty/master/patches/nginx-1.17.4-stream_proxy_timeout_fields.patch
+    wget -P patches https://raw.githubusercontent.com/openresty/openresty/master/patches/nginx-1.17.4-stream_ssl_preread_no_skip.patch
+    wget -P patches https://raw.githubusercontent.com/openresty/openresty/master/patches/nginx-1.17.4-upstream_pipelining.patch
+    wget -P patches https://raw.githubusercontent.com/openresty/openresty/master/patches/nginx-1.17.4-upstream_timeout_fields.patch
```
Comment (membphis) [inline .travis/linux_tengine_runner.sh]: Can it be written as an array here?

Comment (membphis) [reply .travis/linux_tengine_runner.sh]: Avoid duplication of code here

File: .travis/linux_tengine_runner.sh [hunk 2]
```diff
@@ -68,6 +68,65 @@ tengine_install() {
...
+    patch -p1 < ../../patches/nginx-1.17.4-safe_resolver_ipv6_option.patch
+    patch -p1 < ../../patches/nginx-1.17.4-single_process_graceful_exit.patch
+    patch -p1 < ../../patches/nginx-1.17.4-ssl_cert_cb_yield.patch
+    patch -p1 < ../../patches/nginx-1.17.4-ssl_sess_cb_yield.patch
+    patch -p1 < ../../patches/nginx-1.17.4-stream_balancer_export.patch
+    patch -p1 < ../../patches/nginx-1.17.4-stream_proxy_get_next_upstream_tries.patch
+    patch -p1 < ../../patches/nginx-1.17.4-stream_proxy_timeout_fields.patch
+    patch -p1 < ../../patches/nginx-1.17.4-stream_ssl_preread_no_skip.patch
+    patch -p1 < ../../patches/nginx-1.17.4-upstream_pipelining.patch
+    patch -p1 < ../../patches/nginx-1.17.4-upstream_timeout_fields.patch
```
Comment (moonming) [inline .travis/linux_tengine_runner.sh]: Is there a way to make sure these patches are working properly?

Comment (totemofwolf) [reply .travis/linux_tengine_runner.sh]: Test cases exit without errors. I will post an issue to Alibaba tengine right now.

Comment (totemofwolf) [reply .travis/linux_tengine_runner.sh]: Here it is : [https://github.com/alibaba/tengine/issues/1397](url)

Comment (moonming) [reply .travis/linux_tengine_runner.sh]: got it. we can wait for tengine's response beofre merge.

File: .travis/linux_tengine_runner.sh (hunk 1 above)
Comment (totemofwolf) [reply .travis/linux_tengine_runner.sh]: > Avoid duplication of code here

Duplication will makes patch errors ;-) and hang the ci process.

Comment (membphis) [reply .travis/linux_tengine_runner.sh]: @totemofwolf I mean this way: https://stackoverflow.com/questions/8880603/loop-through-an-array-of-strings-in-bash

```
$ cat test.sh
for Item in Item1 Item2 Item3 Item4 ;
  do
    echo $Item
  done

$ sh test.sh
Item1
Item2
Item3
Item4
```

File: .travis/linux_tengine_runner.sh (hunk 2 above)
Comment (totemofwolf) [reply .travis/linux_tengine_runner.sh]: > got it. we can wait for tengine's response beofre merge.

--
@totemofwolf 如果使用到lua-resty-core里面的一些特殊功能的话，是需要打上这些补丁的。
--

via: 
https://github.com/alibaba/tengine/issues/1397#issuecomment-553883919

Comment (membphis) [reply .travis/linux_tengine_runner.sh]: > got it. we can wait for tengine's response beofre merge.

Tengine hasn't decided to reference the lua-resty-core file yet, and our compilation method is more like OpenResty.

I suggest that we can use it in travis first to see if we can find new problems.

File: .travis/linux_tengine_runner.sh (hunk 1 above)
Comment (membphis) [reply .travis/linux_tengine_runner.sh]: @totemofwolf 

enable privileged process feature:
https://github.com/openresty/openresty/blob/master/patches/nginx-1.17.4-privileged_agent_process.patch

enable 
https://github.com/openresty/openresty/blob/master/patches/nginx-1.17.4-resolver_conf_parsing.patch

File: .travis/linux_tengine_runner.sh (hunk 2 above)
Comment (totemofwolf) [reply .travis/linux_tengine_runner.sh]: Got it. I will also use it in our qa env first.

**Pull request merged**